walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,100 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
+  /**
+   * Returns a boolean based on if at least one participant in the event is also a participant in
+   * the meeting request.
+   *
+   * @param eventAttendeesCopy A copy of the set of attendees in a specific event.
+   * @param meetingAttendees The set of attendees in the meeting request.
+   * @return True if at least one participant in the set {@code eventAttendees} is also a
+   *     participant in {@code meetingAttendees}. False otherwise.
+   */
+  private boolean eventParticipantInMeeting(
+      Set<String> eventAttendeesCopy, Collection<String> meetingAttendees) {
+    eventAttendeesCopy.retainAll(meetingAttendees);
+    return !eventAttendeesCopy.isEmpty();
+  }
+
+  /**
+   * Returns a collection of all meeting times that will work for all attendees' schedules.
+   *
+   * <p>The open meeting times are determined by iterating through each event and creating a new
+   * {@code TimeRange} for each range between meetings. If the duration is greater than or equal to
+   * the duration of the requested meeting, that {@code TimeRange} instance is added to the list
+   * {@code openMeetingTimes}. The initial value of {@code endOfEarlierEvent} is set to the start of
+   * the day for convenience when evaluating the first event. Also, the end of day time is appended
+   * as an additional event to the current event list to avoid more conditional statements.
+   *
+   * @param eventList The list of events that are used to determine what periods of time that the
+   *     meeting can take place. This list is 'filtered' and sorted in {@code query()}.
+   * @param request The meeting request that contains the requirements for potential meetings.
+   * @return A Collection of the feasible meeting {@code TimeRange}s.
+   */
+  private Collection<TimeRange> getMeetingTimes(List<Event> eventList, MeetingRequest request) {
+    eventList.add(new Event(
+        "EOD", TimeRange.fromStartDuration(TimeRange.END_OF_DAY, 0), request.getAttendees()));
```

Comment:
Since EOD is used multiple times, it should probably be separated out into a variable.

Reply:
Done.